design/batch_retrieval_design.md

```diff
@@ -1,20 +1,215 @@
+*Now*, with the Job Milestone, the Secrets Provider is deployed outside of the customers' Pods, supporting multiple customer pods. With this, a single batch retrieval request will be made for *all* applications to fetch the necessary secrets to deploy properly. Therefore, if there is a failure in receiving one of the values, then the whole request will fail. In other words, **no application** Pod will spin up in the namespace. This leads to a faulty user experience.
 
-|      | Solution                                                     | Pros                                                         | Cons                                                         | Effort (T-shirt size) |
-| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | --------------------- |
-| 1    | *Client side:* <br />For each K8s Secret, perform a new Batch request | - No server side changes / no breaking changes               | Load on server with extra calls (*)                          | S, 5 days             |
-| 2    | *Server side:*<br />Update Batch retrieval to return list of variables **and** their response (success/failure) (**) | - Will help us during rotation for Milestone 2<br />- Better / straightforward design for how batch endpoints | - Requires both client/server implementation<br />- Need to handle backwards compatibility | M, 10 days            |
-| 3    | Stay as is                                                   | No additional work needed                                    | Bad UX                                                       | -                     |
+The following table outlines possible solutions to overcome this:
+
+|      | Solution                                                     | Pros                                                         | Cons                                               | Effort (T-shirt size) |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | -------------------------------------------------- | --------------------- |
+| 1    | *Client side:* <br />For each K8s Secret, perform a new Batch request | - No server side changes / no breaking changes               | Load on server with extra calls (*)                | S, 5 days             |
+| 2    | *Server side:*<br />Update Batch retrieval to return list of variables **and** their response (success/failure) (**) | - Will help us during rotation for Milestone 2<br />- Best practice / straightforward design for batch endpoint | - Requires both client/server implementation<br /> | M, 10 days            |
+| 3    | Stay as is                                                   | No additional work needed                                    | - Bad UX<br />- Performance issues                 | -                     |
```

Comment:
> "Load on server with extra calls"

Do we have any data indicating that this would be a problem?  In _theory_ I understand it could, but it's worth making sure we aren't optimizing for a problem that won't exist in practice.  

To be clear I'm not saying the problem wouldn't exist... I'm just asking: How do we know?

Reply:
Also a theory which doesn't seem too convincing but at this time, hasn't been tested so I will just remove that